libpromises/promises.c

```diff
@@ -629,7 +629,7 @@ Promise *ExpandDeRefPromise(EvalContext *ctx, const Promise *pp, bool *excluded)
         if (ifvarclass && !IsVarClassDefined(ctx, ifvarclass, pcopy))
         {
             Log(LOG_LEVEL_VERBOSE, "Skipping promise '%s'"
-                " because 'if'/'ifvarclass' is not defined", pp->promiser);
+                " because it is not defined in this context.", pp->promiser);
 
```

Comment:
The promise is not part of the current context because the if/ifvarclass constraint was not met correct?

The if/ifvarclass constraint may not be met because it contains a variable that's not defined, or because it evaluates to non true.

Reply:
Yes I believe this is correct.  Tracing through the code, I got to the IsDefinedClass in libpromises/eval_context.c, and that returns false because of the final line:

return r == true;